src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -137,21 +153,30 @@ private class ACCPService extends Service {
 
         public ACCPService(
                 final String type, final String algorithm, final String className,
+                final boolean useReflection, // this flag determines if the instantiation of the service class is supposed to be done via reflection or explicitly
```

Comment:
In general, it's better not to use reflection due to its performance overhead. Another reason to not use reflection is our `addService` method assumes the constructor of the SPI class does not take any argument; in case an SPI's constructor has arguments, it's best to explicitly instantiate it with the right arguments.